### PR TITLE
[Merged by Bors] - feat(Filter): lemmas from Mandelbrot connectedness project

### PR DESCRIPTION
**Pull request by urkud**
- Add `Filter.frequently_iff_neBot`
  and `Filter.frequently_mem_iff_neBot`.

- Drop some implicit arguments
  that are available from `variable`.

- Add `Filter.disjoint_prod` and `Filter.frequently_prod_and`.

- Swap `Filter.le_prod` with `Filter.tendsto_prod_iff'`
  to use the latter in the proof of the former.

Co-Authored-By: @girving

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by girving** [comment]
Thank you for doing this!  And the golfing in `ray` looks great too; I'll go through and remove some of the now-trivial ones this evening or tomorrow.

File: Mathlib/Order/Filter/Basic.lean [hunk 1]
```diff
@@ -1310,6 +1310,12 @@ theorem Eventually.exists {p : α → Prop} {f : Filter α} [NeBot f] (hp : ∀
   hp.frequently.exists
 #align filter.eventually.exists Filter.Eventually.exists
 
+lemma frequently_iff_neBot {p : α → Prop} : (∃ᶠ x in l, p x) ↔ NeBot (l ⊓ 𝓟 {x | p x}) := by
```
Comment (ADedecker) [inline Mathlib/Order/Filter/Basic.lean:1313]: Could you golf `frequently_true_iff_neBot` using this?

**Review by ADedecker — APPROVED** [review]
Thanks!

bors d+

**Comment by urkud** [comment]
bors merge

**Pull request closed without merging**